crates/ln-dlc-node/src/node/connection.rs

```diff
@@ -40,16 +48,20 @@ impl Node {
     }
 
     pub async fn keep_connected(&self, peer: NodeInfo) -> Result<()> {
-        // TODO: Let this time out
-        let connection_closed_future = loop {
-            tracing::debug!(%peer, "Attempting to establish initial connection");
+        #[allow(clippy::async_yields_async)] // we want to poll the returned future in a loop
+        let connection_closed_future = tokio::time::timeout(Duration::from_secs(30), async {
```

Comment:
🔧 I am not in favor of this, as I think it makes sense for the app to indefinitely try to reconnect to the coordinator. If I read this correctly the behavior would change towards the app only tries 30 seconds after loosing the connection and would then end up in a failed state.

Reply:
This is only with respect to the initial connection. It was designed to only return after establishing the initial connection, which is why it can block execution indefinitely.

In practice, the feature of waiting for the initial connection does not matter for the only production user (the app) because we spawn a task to run `keep_connected` anyway.

That means the feature is only useful for the tests. But it is debatable if it is even necessary, as we could just call `connect_to_peer`, since we _shouldn't_ need to reconnect in the tests.

Reply:
Your comment does make me think that maybe the `keep_connected` code should be in the `app`. For now I think I will revert this part of the change and add another patch which:

- Replaces `keep_connected` with `connect_to_peer` in the tests.
- Rewrites `keep_connected` to not do anything special with the initial connection and simply keep the connection alive.

Reply:
Moved these unrelated changes to https://github.com/get10101/10101/pull/370.